hub/core/chunk_engine.py

```diff
@@ -563,8 +565,12 @@ def _samples_to_chunks(
         nsamples = len(samples)
         if register and update_commit_diff:
             commit_diff = self.commit_diff
+        if progressbar:
+            pbar = tqdm(total=len(samples))
         while len(samples) > 0:
             num_samples_added = current_chunk.extend_if_has_space(samples)  # type: ignore
+            if progressbar:
+                pbar.update(num_samples_added)
```

Comment:
this only updates the progress bar in chunks. would be better to have updates per sample.

Reply:
While I understand that updating the progress bar per sample would make it more informative of how many samples have been uploaded in total, wouldn't updating it per chunk also provide the same information? The progress bar runs only for the total number of samples anyway.

If it still needs to be changed, could you please point me in the right direction? I can't see a way of updating it sample wise as the updation itself is being done in chunks.